packages/xdl/src/Webpack.ts

```diff
@@ -628,8 +628,10 @@ async function loadConfigAsync(
   argv?: string[]
 ): Promise<WebpackConfiguration> {
   setMode(env.mode);
-  // Check if the project has a webpack.config.js in the root.
-  const projectWebpackConfig = path.resolve(env.projectRoot, 'webpack.config.js');
+  // Check if the project has a webpack.config.{ts,js} in the root.
+  const projectWebpackConfig = ["ts", "js"]
+    .map(ext => path.resolve(env.projectRoot, `webpack.config.${ext}`))
+    .find(filePath => fs.existsSync(filePath));
   let config: WebpackConfiguration;
   if (fs.existsSync(projectWebpackConfig)) {
     const webpackConfig = require(projectWebpackConfig);
```

Comment:
This line won't support TypeScript, perhaps try testing it with some types in the file.

Reply:
You are right. The following is the error I encounter when types exist in `webpack.config.ts`. I am happy to close this PR unless you see a solution?

![image](https://user-images.githubusercontent.com/170023/133162081-a41643a0-4236-4dd9-b2b8-5e2e54649aa9.png)

Reply:
I extended the code to [use `ts-node` to patch `require()`](https://github.com/TypeStrong/ts-node#programmatic) to transpile TypeScript if using the TypeScript version. This avoids the syntax error shown above.